### ISSUE TITLE
Missing diagnostics messages

### ISSUE THREAD

**Issue by robb-j** [eclipse-lemminx/lemminx#1509]
Hi, I've been working on an [XML Extension](https://extensions.panic.com/extensions/robb-j/robb-j.XML/) that uses this Language Server and have encountered an issue.

I only seem to get an initial `textDocument/publishDiagnostics` when opening a document and no following ones after the document is edited. I've been using the binary released with [redhat-developer/vscode-xml](github.com/redhat-developer/vscode-xml/), currently using their `0.24.0` build, I tried a few other versions but the same issue occurs.

I've got the LSP log from Nova and a screencast of what I was doing while it was captured:

[lsp.log](https://github.com/eclipse/lemminx/files/11186297/lsp.log)

https://user-images.githubusercontent.com/1526174/230796358-11d05a6b-51b7-4a23-9122-dc05fed53e34.mov

I'm not sure where to start debugging this, I was expecting a `textDocument/publishDiagnostics` notification after the `textDocument/didChange` to clear the error message? Any help/guidance would be appreciated.

**Comment by robb-j** [comment]
Adding another log with the "initialise" messages in it

[lsp.log](https://github.com/eclipse/lemminx/files/11186316/lsp.log)

**Comment by angelozerr** [comment]
Hvae you this problem with another file? Is it possible to share your XML and XSD please.

**Comment by angelozerr** [comment]
Is it working with vscode-xml?

**Comment by robb-j** [comment]
It seems to be for any file not just that specific one. The example above was the [examples/KitchenSink.xml](https://github.com/robb-j/nova-xml/tree/main/examples) from my extension repo. The same issue happens with `Inventory.xml` and `Inventory.xsd` in that same folder which is a simpler reproduction.

Both files validate fine in `vscode-xml` and the diagnostics go away once fixed.

**Comment by angelozerr** [comment]
> Both files validate fine in vscode-xml and the diagnostics go away once fixed.

If I understand your comment, it is working on vscode-xml? If it that it is a problem with nova LSP client.

**Comment by robb-j** [comment]
Yes, something is up, either in Nova or Lemminx. It seems the server isn't sending a `textDocument/publishDiagnostics` message after a `textDocument/didChange` is sent to it, so the errors don't go away.

I got a debugging version running on my local machine and there an error in `TextDocumentContentChangeEvent#getRangeLength` where the "range" that Nova is sending is `null` . From [the spec](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#textDocumentContentChangeEvent) thats allowed but maybe it should be undefined rather than null?

This is the code that is calling it
https://github.com/eclipse/lemminx/blob/b34a847c51043ccf782da8646d25323a1fa290ab/org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/commons/TextDocument.java#L169-L171
 
Here's the error log: [lsp4xml.log](https://github.com/eclipse/lemminx/files/11202840/lsp4xml.log)

If you can point me in the right direction I'm more than happy to submit a PR if it is an issue with the server, or I can take this to Nova and submit a report there if it isn't being spec-compliant?
